mesonbuild/mlog.py

```diff
@@ -370,10 +375,10 @@ def format_list(input_list: T.List[str]) -> str:
         return ''
 
 @contextmanager
-def nested() -> T.Generator[None, None, None]:
+def nested(name: str = '') -> T.Generator[None, None, None]:
     global log_depth
-    log_depth += 1
+    log_depth.append(name)
     try:
         yield
     finally:
-        log_depth -= 1
+        log_depth.pop()
```

Comment:
nit: pop is for returning a value, `del log_depth[-1]` is more correct.

Reply:
Does it really change anything? IMHO pop() is more readable.